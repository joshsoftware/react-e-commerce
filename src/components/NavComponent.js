import React from 'react';
import NavItemComponent from './NavItemComponent';
import UncontrolledDropdownComponent from './UncontrolledDropdownComponent';
import cart from '../images/cart.jpg';
import { Nav } from 'reactstrap';

const NavComponent = () => {
  return (
    <>
      <Nav className="mr-auto" navbar>
        <NavItemComponent url={'/components'} lable={'Shop'} />
        <NavItemComponent url={'/about'} lable={'About'} />
      </Nav>
      <Nav>
        <NavItemComponent url={'/about'} logo={cart} />
        <UncontrolledDropdownComponent />
      </Nav>
    </>
  );
};

export default NavComponent;
