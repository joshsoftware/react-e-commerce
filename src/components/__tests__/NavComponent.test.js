import NavComponent from '../NavComponent';
import { render } from '@testing-library/react';
import React from 'react';

describe('Nav Component', () => {
  it('Must return Nav Component', () => {
    const { asFragment } = render(<NavComponent />);

    expect(asFragment()).toMatchSnapshot();
  });
});
