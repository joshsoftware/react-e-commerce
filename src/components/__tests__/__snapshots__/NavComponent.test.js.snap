// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component Must return Nav Component 1`] = `
<DocumentFragment>
  <ul
    class="mr-auto navbar-nav"
  >
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="/components"
      >
        Shop
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="/about"
      >
        About
      </a>
    </li>
  </ul>
  <ul
    class="nav"
  >
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="/about"
      >
        <img
          alt="img"
          class="rounded-circle"
          src="cart.jpg"
          style="width: 70px;"
        />
      </a>
    </li>
    <li
      class="dropdown nav-item"
    >
      <a
        aria-expanded="false"
        aria-haspopup="true"
        class="dropdown-toggle nav-link"
        href="#"
      >
        <img
          alt="img"
          class="rounded"
          src="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
          style="width: 50px;"
        />
      </a>
      <div
        aria-hidden="true"
        class="dropdown-menu dropdown-menu-right"
        role="menu"
        tabindex="-1"
      >
        <button
          class="dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          Profile
        </button>
        <div
          class="dropdown-divider"
          tabindex="-1"
        />
        <button
          class="dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          Logout
        </button>
        <div
          class="dropdown-divider"
          tabindex="-1"
        />
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
